File: modules/live-activity/src/LiveActivityModule.ts
import { requireOptionalNativeModule } from 'expo-modules-core';

import type {
  LiveActivityHandle,
  LiveActivityStart,
  LiveActivityUpdate,
} from './types';

type LiveActivityModuleType = {
  isAvailable(): boolean;
  startActivity(input: LiveActivityStart): Promise<LiveActivityHandle | null>;
  updateActivity(activityId: string, update: LiveActivityUpdate): Promise<void>;
  endActivity(activityId: string): Promise<void>;
};

// Optional: null when not linked (Expo Go / web / Jest).
export default requireOptionalNativeModule<LiveActivityModuleType>(
  'HoneLiveActivityModule',
);

File: modules/live-activity/android/src/main/java/expo/modules/liveactivity/LiveActivityModule.kt
package expo.modules.liveactivity

import expo.modules.kotlin.Promise
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * Android stub. ActivityKit is iOS-only. Android equivalents (Quick Tiles,
 * MediaSession ongoing notification) are out of scope for v1.1.
 */
class LiveActivityModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("HoneLiveActivityModule")

    Function("isAvailable") { -> false }

    AsyncFunction("startActivity") { _input: Map<String, Any?>, promise: Promise ->
      promise.resolve(null)
    }

    AsyncFunction("updateActivity") { _id: String, _update: Map<String, Any?>, promise: Promise ->
      promise.resolve(null)
    }

    AsyncFunction("endActivity") { _id: String, promise: Promise ->
      promise.resolve(null)
    }
  }
}
